Fixing some of the color correction math

tradiuz opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->



## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Lower mireds for max (cool) color temp 500 -> 370  (changes lower color temp from 2000K to 2700K, which is more in line with most common CCT bulbs)
I'd love to add a variable to adjust the max/min per bulb (in case that bulb supports amber/UV to give fuller range), but that is out of my depth.

Also added limit math to the color temp calculation to prevent overflow/underflow (if you provide 1000K or 9000K color temp values, it just gives you the min/max, not random numbers.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests


## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue:  This continues @firstof9 's work (PR#37636) on the ozw integration and rgb lights 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [X] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

tradiuz committed:
Fixing some of the color correction math 

Mireds 500 -> 370  (changes lower color temp from 2000K to 2700K, which is more in line with most common CCT bulbs)
I'd love to add a variable to adjust the max/min per bulb (in case that bulb supports amber/UV to give fuller range), but that is out of my depth.


Added limit math to the color temp calculation to prevent overflow/underflow (if you provide 1000K or 9000K color temp values, it just gives you the min/max, not random numbers)

MartinHjelmare commented on homeassistant/components/ozw/light.py:
```diff
@@ -194,9 +194,11 @@ async def async_turn_on(self, **kwargs):
 
         elif color_temp is not None:
             cold = round((TEMP_COLOR_MAX - round(color_temp)) / TEMP_COLOR_DIFF * 255)
+            if cold < 0:
```
Please use `min` and `max` instead.

MartinHjelmare reviewed (changes requested):
Please make sure the build passes.

tradiuz committed:
Editing color temp limits and fixing comments

Cleaning up if/thens and using min/max per @MartinHjelmare

Adding a space for inline comments.

marcelveldt commented:
@tradiuz why did you close the PR ? Seems to me that these changes are very valid.

firstof9 commented:
Reopened as https://github.com/home-assistant/core/pull/38012
